R/output_format.R

```diff
@@ -103,10 +103,7 @@ merge_post_processors <- function (base, overlay) {
 merge_output_formats <- function(base, overlay)  {
   structure(list(
     knitr = merge_lists(base$knitr, overlay$knitr),
-    pandoc = pandoc_options(
-      to = merge_scalar(base$pandoc$to, overlay$pandoc$to),
-      from = merge_scalar(base$pandoc$from, overlay$pandoc$from),
-      args = c(base$pandoc$args, overlay$pandoc$args)),
+    pandoc = merge_lists(base$pandoc, overlay$pandoc),
```

Comment:
You're right, I missed that. The issue with the current code is that only a subset of pandoc options is captured, with `keep_tex`, `latex_engine` and `ext` being discarded. I discovered this when implementing a format extending the base rmarkdown `pdf_document`.

I've reverted commit 49ea937 and extended the original approach in e4cf95e.